clients/Barracks.js

```diff
@@ -677,6 +680,27 @@ class Barracks {
       });
     });
   }
+
+  getDeploymentPlan(token, componentRef) {
+    return new Promise((resolve, reject) => {
+      logger.debug('Getting DeploymentPlan for component', componentRef);
+      this.client.sendEndpointRequest('getDeploymentPlan', {
+        headers: {
+          'x-auth-token': token
+        },
+        pathVariables: {
+          componentRef
+        }
+      }).then(response => {
+        const deploymentPlan = response.body;
+        logger.debug('DeploymentPlan retrieved:', deploymentPlan);
+        resolve(deploymentPlan);
+      }).catch(errResponse => {
```

Comment:
Rename variable to err, it's not always an error from the response, it could be execution failling